pkg/block/azure/client_cache.go

```diff
@@ -88,19 +92,18 @@ func (c *ClientCache) NewServiceClient(storageAccount string) (*service.Client,
 }
 
 func (c *ClientCache) NewUDC(ctx context.Context, storageAccount string, expiry *time.Time) (*service.UserDelegationCredential, error) {
-	// Otherwise assume using role based credentials and build signed URL using user delegation credentials
-	currentTime := time.Now().UTC().Add(-10 * time.Second)
-	// UDC expiry time of PreSignedExpiry + hour
-	udcExpiry := expiry.Add(time.Hour)
-	info := service.KeyInfo{
-		Start:  to.Ptr(currentTime.UTC().Format(sas.TimeFormat)),
-		Expiry: to.Ptr(udcExpiry.Format(sas.TimeFormat)),
-	}
-
 	var udc *service.UserDelegationCredential
 	// Check udcCache
 	res, ok := c.udcCache.Get(storageAccount)
 	if !ok {
+		// Otherwise assume using role based credentials and build signed URL using user delegation credentials
+		currentTime := time.Now().UTC().Add(-10 * time.Second)
+		// UDC expiry time of PreSignedExpiry + hour
+		udcExpiry := expiry.Add(UDCCacheExpiry)
```

Comment:
What is the maximum UDC expiry that Azure will give us?  For instance, if the user limits our UDC expiry to 1 hour, then the request below will always fail because it is for >1 hour.

Reply:
User does not control the UDC expiry - it is used internally only for caching purposes

Reply:
This is odd.  I would expect Azure to be more like AWS and even other MS products, and allow clients to request almost any expiry time but then trim that according to some expiration.

Reply:
The client in this case is lakeFS and not the user - and there's a distinction between the UDC expiry time and the presigned URL expiry